### PR TITLE
Remove feeMake from contract since it is 0 anyways

### PR DESCRIPTION
**Pull request by JonathonDunford**
Add node files to gitignore

### CODE REVIEW

**Comment by mmhh1910** [comment]
I agree that makeFee is no longer needed, but I would probably not change the interface (the public functions and properties) of the contract, since it will break existing code calling the ED/FD contract.

**Comment by JonathonDunford** [comment]
@betterdelta 

How would it change the interface?

`makeFee` was only touched inside the smart contract, not outside.

File: contracts/ForkDelta.sol [hunk 1]
```diff
@@ -34,10 +33,9 @@ contract ForkDelta {
   }
 
   /// Constructor function. This is only called on contract creation.
-  function ForkDelta(address admin_, address feeAccount_, uint feeMake_, uint feeTake_, uint freeUntilDate_) public {
+  function ForkDelta(address admin_, address feeAccount_, uint feeTake_, uint freeUntilDate_) public {
```
Comment (Ebonsignori) [inline contracts/ForkDelta.sol:36]: Here you change the function signature. Will this effect the API that interacts with the contract? 

Nevermind, this comment answers my question
"//... This is only called on contract creation."

**Comment by Ebonsignori** [comment]
After ctrl+f and slow read of each feeMake or make appearance, I'm confident that you didn't miss anything. To answer @betterdelta's concern, the only public function changed was the signature of ForkDelta() by removing the feeMake argument. However, this is only used for contract construction, so everything related to the interface is untouched

**Review by Ebonsignori — APPROVED** [review]
@betterdelta the only altered public functions are: ForkDelta and changeFeeMake. 

ForkDelta: Since this is the constructor function that is only called on contract creation, the interface for interacting with the contract appears untouched. 

changeFreeMake: This is an administrative function with an isAdmin modifier, and so it also doesn't effect the existing code interfacing. 

I didn't find any missing occurrences of make after scanning the code. Seems legit. 

:shipit:

File: contracts/ForkDelta.sol [hunk 2]
```diff
@@ -58,12 +56,6 @@ contract ForkDelta {
     feeAccount = feeAccount_;
   }
 
-  /// Changes the fee on makes. Can only be changed to a value less than it is currently set at.
```
Comment (Ebonsignori) [inline contracts/ForkDelta.sol:61]: While this is also a changed signature, it was an administrative one that the ForkDelta API doesn't directly interact with

**Comment by mmhh1910** [comment]
Sorry for being unclear. For me the public variables are part of the interface and feeMake is one of them:

https://etherscan.io/address/0x8d12a197cb00d4747a1fe03395095ce2a5cc6819#readContract

This will break existing code, e.g. trade bots might have a calculation if a trade is profitable or what the cost of a trade are and read the public variable feeMake for it.

**Comment by JonathonDunford** [comment]
@betterdelta 

Bots are going to break anyways as they will be pointing to the old contract. They will have to make updates to point them to the new contract and they will have to redownload the contract ABI.

I think this is a non-issue. As long as we are not breaking our interface, we will be fine.

Also, this should be V1 of our contract. We shouldn't let it suffer because of backwards support for bots especially if they will have to make edits anyways.

**Comment by mmhh1910** [comment]
If we don't change the interface (or at least do not remove anything) the only change that is needed is a contract address update, which is more of a configuration change. ABI does not have to be updated as long as the we don't remove or change public functions/variables. 

If you don't think it is an issue that the contract will not be backward compatible, go ahead. 

From the code point of view I see no problems with this PR.

**Comment by Ebonsignori** [comment]
@betterdelta 

You make a good point, the ABI will change for anything external of FD that interfaces with our smart contract. 

First, those external things will still have to update to the contract's new address, and when they do that there is nothing stopping them from updating their ABI as well. 

Second, an ABI that doesn't cover every function shouldn't matter. As far as Web3 (and I'm sure other libraries) goes, you can work with ABI's that don't cover all of a contract's interface. Consider [this problem](https://ethereum.stackexchange.com/questions/29914/getting-abi-data-for-erc20-tokens-programatically-with-web3) . You can use a generic ABI that has only common ERC20 functions and access those functions with an "incomplete" ABI.

**Pull request merged**